maggy/core/environment_singleton.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+"""
+return an instance of the environment to be used by maggy within a session.
+"""
+
+
+def environment_singleton():
+    global environmentInstance
+    if not "environmentInstance" in globals():
```

Comment:
```suggestion
    if not "environment_instance" in globals():
```
use snake_case instead of camelCase in Python.